src/main/java/org/apache/cassandra/sidecar/CassandraSidecarDaemon.java

```diff
@@ -97,6 +101,20 @@ private void validate()
 
     }
 
+    /**
+     * Checks the health of every instance configured in the {@link Configuration#getInstancesConfig()}.
+     * The health check is executed in a blocking thread to prevent the event-loop threads from blocking.
+     *
+     * @param timerId the ID of the periodic timer
+     */
+    private void healthCheck(Long timerId)
+    {
+        config.getInstancesConfig()
+              .instances()
+              .forEach(instanceMetadata ->
+                       vertx.executeBlocking(promise -> instanceMetadata.delegate().healthCheck()));
```

Comment:
If my understanding is correct. 
In this setup, if the prior check takes a longer time that is greater than the interval, there could be 2 threads that runs `healthCheck` concurrently. Due to the check is `synchronized`, there will be one thread being blocked. 
In the prior setup, no additional thread is to be blocked. 

I do not mean that the prior one is better.

One step back, the health check is something we want to perform every X seconds. If a prolonged check just finished, the next one should not be immediately performed, which is the problem for both setups. 

Can we add a `shouldCheck(timeInMillis)` method and skip a check if it has just checked very recently?